### PR TITLE
Rename script to removePII

### PR DESCRIPTION
**Pull request by paladox**

### CODE REVIEW

File: maintenance/removePII.php [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program; if not, write to the Free Software Foundation, Inc.,
+* 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+* http://www.gnu.org/copyleft/gpl.html
+*
+* @file
+* @ingroup Maintenance
+* @author Miraheze Site Reliability Engeneering team
```
Comment (Southparkfan) [inline maintenance/removePII.php:21]: Eng**i**neering

File: maintenance/removePII.php [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+		parent::__construct();
+		$this->mDescription = "Removes PII information from users (e.g email addresses, ip address and other identifying information).";
+		$this->addOption( 'delete', 'Deletes PII info', false, false );
+		$this->addOption( 'username', 'Username to be used to find requested information.', true, true );
+	}
+
+	public function execute() {
+		global $wgCentralAuthDatabase;
+
+		if ( !(bool)$this->getOption( 'delete' ) ) {
```
Comment (Southparkfan) [inline maintenance/removePII.php:38]: What's the purpose of the --delete switch if nothing (other than 'global $wgCentralAuthDatabase;') happens?

File: maintenance/removePII.php [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+		$user = User::newFromName( (string)$this->getOption( 'username' ) );
+		if ( !$user ) {
+			$this->output( 'User does not exist' );
+			return;
+		}
+
+		$dbw = wfGetDB( DB_MASTER );
+
+		if ( class_exists( 'AJAXPoll' ) ) {
```
Comment (Southparkfan) [inline maintenance/removePII.php:50]: An extension may be disabled at a later stage, so the class_exists check would return false, even if the data is present nevertheless. An alternative could be using $dbw->tableExists( 'name_of_table' ); before attempting to update the table contents.

File: maintenance/removePII.php [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+			$dbw->update(
+				'ajaxpoll_vote',
+				[
+					'poll_ip' => ''
+				],
+				[
+					'poll_actor' => $user->getActorId( $dbw )
```
Comment (Southparkfan) [inline maintenance/removePII.php:57]: $user->getActorId( $dbw ) is being referenced to many times. Please consider retrieving this once and using a variable for the update functions.

File: maintenance/removePII.php [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+		}
+
+		if ( class_exists( 'Flow' ) ) {
+			$dbw->update(
+				'flow_tree_revision',
+				[
+					'tree_orig_user_ip' => ''
+				],
+				[
+					'tree_orig_user_id' => $user->getId()
```
Comment (Southparkfan) [inline maintenance/removePII.php:83]: $user->getId() is being referenced to many times. Please consider retrieving this once and using a variable for the update functions.

File: maintenance/removePII.php [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+				],
+				[
+					'poll_actor' => $user->getActorId( $dbw )
+				],
+				__METHOD__
+			);
+		}
+
+		if ( class_exists( 'Comment' ) ) {
+			$dbw->update(
```
Comment (Southparkfan) [inline maintenance/removePII.php:64]: Due to the volume of $dbw->update calls, all being very similar, I suggest we make this script more extensible by using a multi-dimensional array. Such as:
```php
[
	'Comments' => [
		'fields' => [
			'Comment_IP' => '',
		],
		'where' => [
			'Comment_actor' => $userActorId,
		],
	],
	'flow_tree_revision' => [
		'fields' => [
			'tree_orig_user_ip' => '',
		],
		'where' => [
			'tree_orig_user_id' => $userId,
		],
	],
]
```

File: maintenance/removePII.php (hunk 1 above)
Comment (paladox) [reply maintenance/removePII.php]: Thanks, done.

File: maintenance/removePII.php (hunk 2 above)
Comment (paladox) [reply maintenance/removePII.php]: Removed, thanks, done.

File: maintenance/removePII.php (hunk 4 above)
Comment (paladox) [reply maintenance/removePII.php]: Thanks, done.

File: maintenance/removePII.php (hunk 5 above)
Comment (paladox) [reply maintenance/removePII.php]: Thanks, done.

File: maintenance/removePII.php (hunk 6 above)
Comment (paladox) [reply maintenance/removePII.php]: Thanks, done.

File: maintenance/removePII.php (hunk 3 above)
Comment (paladox) [reply maintenance/removePII.php]: Thanks, done.

**Pull request merged**